test/text_overlay/text_overlay_native_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule TextOverlay.NativeTest do
+  use ExUnit.Case, async: true
+
+  alias Membrane.FFmpeg.VideoFilter.TextOverlay.Native
+  alias VideoFilter.Helpers
+
+  @tag :tmp_dir
+  test "overlay text over raw video frame", %{tmp_dir: tmp_dir} do
+    {in_path, out_path, ref_path} = Helpers.prepare_paths("1frame.yuv", "ref-native.yuv", tmp_dir)
+
+    assert {:ok, frame} = File.read(in_path)
+
+    assert {:ok, ref} =
+             Native.create(
+               "mtext",
+               640,
+               360,
+               :I420,
+               12,
+               false,
+               "",
+               false,
+               "white",
+               "",
+               :center,
+               :top
+             )
+
+    assert {:ok, out_frame} = Native.apply_filter(frame, ref)
+    assert {:ok, file} = File.open(out_path, [:write])
+
+    IO.binwrite(file, out_frame)
+    File.close(file)
```

Comment:
One more nitpick:
```suggestion
    assert {:ok, file} = File.open(out_path, [:write])
    on_exit(fn -> File.close(file) end)
    
    IO.binwrite(file, out_frame)
```

This will close the file even if `binwrite` crashes